recipes/subunit/all/patches/0001-no-PREFIX-WriteMakefile.patch

```diff
@@ -0,0 +1,16 @@
+'Fixes' this error:
+Only one of PREFIX or INSTALL_BASE can be given.  Not both.
+
+I don't think the fix is correct.
+But perl is not used, so does it matter?
+
+--- perl/Makefile.PL.in
++++ perl/Makefile.PL.in
+@@ -1,6 +1,6 @@
+ use ExtUtils::MakeMaker;
+ WriteMakefile(
+-    'PREFIX' => '@prefix@',
```

Comment:
Omitting PREFIX will install a Perl module to system path (`/System/Library/Perl5` on mac) which is not correct. Where did you see error " Only one of PREFIX or INSTALL_BASE can be given.  Not both. "?

Unforunately, disabling perl module requires deep patches to `configure.ac` and `Makefile.am`. Much easier to install perl modules to package directory and then drop then on packaging.

Reply:
The error is emitted during the `make` step.
I've pasted the complete log:

<details>

<summary>Failed build when dropping the prefix patch</summary>

```
conan create . subunit/1.4.0@
[HOOK - conan-center.py] pre_export(): [DEPRECATED GLOBAL CPPSTD (KB-H001)] OK
[HOOK - conan-center.py] pre_export(): [REFERENCE LOWERCASE (KB-H002)] OK
[HOOK - conan-center.py] pre_export(): [RECIPE METADATA (KB-H003)] OK
[HOOK - conan-center.py] pre_export(): [HEADER_ONLY, NO COPY SOURCE (KB-H005)] OK
[HOOK - conan-center.py] pre_export(): [FPIC OPTION (KB-H006)] OK
[HOOK - conan-center.py] pre_export(): [VERSION RANGES (KB-H008)] OK
[HOOK - conan-center.py] pre_export(): [RECIPE FOLDER SIZE (KB-H009)] Total recipe size: 6.716796875 KB
[HOOK - conan-center.py] pre_export(): [RECIPE FOLDER SIZE (KB-H009)] OK
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] exports: None
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] exports: patches/**
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE FOLDER (KB-H024)] OK
[HOOK - conan-center.py] pre_export(): [META LINES (KB-H025)] OK
[HOOK - conan-center.py] pre_export(): [CONAN CENTER INDEX URL (KB-H027)] OK
[HOOK - conan-center.py] pre_export(): [CMAKE MINIMUM VERSION (KB-H028)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE - RUN ENVIRONMENT (KB-H029)] OK
[HOOK - conan-center.py] pre_export(): [SYSTEM REQUIREMENTS (KB-H032)] OK
[HOOK - conan-center.py] pre_export(): [CONANDATA.YML FORMAT (KB-H030)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE - NO IMPORTS() (KB-H034)] OK
[HOOK - conan-center.py] pre_export(): [NO AUTHOR (KB-H037)] OK
[HOOK - conan-center.py] pre_export(): [NO TARGET NAME (KB-H040)] OK
[HOOK - conan-center.py] pre_export(): [NO FINAL ENDLINE (KB-H041)] OK
[HOOK - conan-center.py] pre_export(): [NO REQUIRES.ADD() (KB-H044)] OK
[HOOK - conan-center.py] pre_export(): [DELETE OPTIONS (KB-H045)] OK
[HOOK - conan-center.py] pre_export(): [NO ASCII CHARACTERS (KB-H047)] OK
[HOOK - conan-center.py] pre_export(): [CMAKE VERBOSE MAKEFILE (KB-H046)] OK
[HOOK - conan-center.py] pre_export(): [CMAKE VERSION REQUIRED (KB-H048)] OK
[HOOK - conan-center.py] pre_export(): [CMAKE WINDOWS EXPORT ALL SYMBOLS (KB-H049)] OK
[HOOK - conan-center.py] pre_export(): [DEFAULT OPTIONS AS DICTIONARY (KB-H051)] OK
[HOOK - conan-center.py] pre_export(): [CONFIG.YML HAS NEW VERSION (KB-H052)] OK
Exporting package recipe
subunit/1.4.0 exports: File 'conandata.yml' found. Exporting it...
subunit/1.4.0 exports: Copied 1 '.yml' file: conandata.yml
subunit/1.4.0 exports_sources: Copied 1 '.patch' file: 0001-no-PREFIX-WriteMakefile.patch
[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] Saving conandata.yml: {'sources': {'1.4.0': {'url': 'https://launchpad.net/subunit/trunk/1.4.0/+download/subunit-1.4.0.tar.gz', 'sha256': '00792096e96cbf1d5f9394dadd0ea73309a4f8b2214822949f1888ce5c2cf1c0'}}}
[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] New conandata.yml contents: sources:
  1.4.0:
    sha256: 00792096e96cbf1d5f9394dadd0ea73309a4f8b2214822949f1888ce5c2cf1c0
    url: https://launchpad.net/subunit/trunk/1.4.0/+download/subunit-1.4.0.tar.gz

[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] OK
[HOOK - conan-center.py] post_export(): [DEFAULT SHARED OPTION VALUE (KB-H050)] OK
subunit/1.4.0: A new conanfile.py version was exported
subunit/1.4.0: Folder: /home/maarten/.conan/data/subunit/1.4.0/_/_/export
subunit/1.4.0: Package recipe modified in export, forcing source folder removal
subunit/1.4.0: Use the --keep-source, -k option to skip it
subunit/1.4.0: Exported revision: ee281034100972853407b4d3da230b9c
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux
os_build=Linux
[options]
[build_requires]
[env]
CC=/usr/bin/gcc
CXX=/usr/bin/g++
subunit/1.4.0: Forced build from source
subunit/1.4.0 (test package): Installing package
Requirements
    cppunit/1.15.1 from 'conan-center' - Cache
    subunit/1.4.0 from local cache - Cache
Packages
    cppunit/1.15.1:b911f48570f9bb2902d9e83b2b9ebf9d376c8c56 - Cache
    subunit/1.4.0:bff1fe8d4fd123419ee54b7d4fcc1b4dea9ea1db - Build

Installing (downloading, building) binaries...
cppunit/1.15.1: Already installed!
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] pre_source(): [IMMUTABLE SOURCES (KB-H010)] OK
subunit/1.4.0: Configuring sources in /home/maarten/.conan/data/subunit/1.4.0/_/_/source

[HOOK - conan-center.py] post_source(): [LIBCXX MANAGEMENT (KB-H011)] OK
[HOOK - conan-center.py] post_source(): [CPPSTD MANAGEMENT (KB-H022)] OK
subunit/1.4.0: Copying sources to build folder
subunit/1.4.0: Building your package in /home/maarten/.conan/data/subunit/1.4.0/_/_/build/bff1fe8d4fd123419ee54b7d4fcc1b4dea9ea1db
subunit/1.4.0: Generator txt created conanbuildinfo.txt
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK. 'fPIC' option found and apparently well managed
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
subunit/1.4.0: Calling build()
subunit/1.4.0: Calling:
 > source_subfolder/configure '--enable-shared=no' '--enable-static=yes' 'CHECK_CFLAGS='\'' '\''' 'CHECK_LIBS='\'' '\''' 'CPPUNIT_CFLAGS='\''-I/home/maarten/.conan/data/cppunit/1.15.1/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/include'\''' 'CPPUNIT_LIBS='\''cppunit'\''' '--prefix=/home/maarten/.conan/data/subunit/1.4.0/_/_/package/bff1fe8d4fd123419ee54b7d4fcc1b4dea9ea1db' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' '--datarootdir=${prefix}/share' 

----Running------
> source_subfolder/configure '--enable-shared=no' '--enable-static=yes' 'CHECK_CFLAGS='\'' '\''' 'CHECK_LIBS='\'' '\''' 'CPPUNIT_CFLAGS='\''-I/home/maarten/.conan/data/cppunit/1.15.1/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/include'\''' 'CPPUNIT_LIBS='\''cppunit'\''' '--prefix=/home/maarten/.conan/data/subunit/1.4.0/_/_/package/bff1fe8d4fd123419ee54b7d4fcc1b4dea9ea1db' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' '--datarootdir=${prefix}/share' 
-----------------
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for style of include used by make... GNU
checking for gcc... /usr/bin/gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /usr/bin/gcc accepts -g... yes
checking for /usr/bin/gcc option to accept ISO C89... none needed
checking whether /usr/bin/gcc understands -c and -o together... yes
checking dependency style of /usr/bin/gcc... gcc3
checking how to run the C preprocessor... /usr/bin/gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking for gcc... (cached) /usr/bin/gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether /usr/bin/gcc accepts -g... (cached) yes
checking for /usr/bin/gcc option to accept ISO C89... (cached) none needed
checking whether /usr/bin/gcc understands -c and -o together... (cached) yes
checking dependency style of /usr/bin/gcc... (cached) gcc3
checking whether we are using the GNU C++ compiler... yes
checking whether /usr/bin/g++ accepts -g... yes
checking dependency style of /usr/bin/g++... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking whether ln -s works... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for fgrep... /usr/bin/grep -F
checking for ld used by /usr/bin/gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from /usr/bin/gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if /usr/bin/gcc supports -fno-rtti -fno-exceptions... no
checking for /usr/bin/gcc option to produce PIC... -fPIC -DPIC
checking if /usr/bin/gcc PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/gcc static flag -static works... yes
checking if /usr/bin/gcc supports -c -o file.o... yes
checking if /usr/bin/gcc supports -c -o file.o... (cached) yes
checking whether the /usr/bin/gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... /usr/bin/g++ -E
checking for ld used by /usr/bin/g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the /usr/bin/g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for /usr/bin/g++ option to produce PIC... -fPIC -DPIC
checking if /usr/bin/g++ PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/g++ static flag -static works... yes
checking if /usr/bin/g++ supports -c -o file.o... yes
checking if /usr/bin/g++ supports -c -o file.o... (cached) yes
checking whether the /usr/bin/g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for python... /usr/bin/python
checking for python version... 2.7
checking for python platform... linux2
checking for python script directory... ${prefix}/lib/python2.7/site-packages
checking for python extension module directory... ${exec_prefix}/lib64/python2.7/site-packages
checking for stdlib.h... (cached) yes
checking for an ANSI C-conforming const... yes
checking for pid_t... yes
checking for size_t... yes
checking whether time.h and sys/time.h may both be included... yes
checking whether struct tm is in sys/time.h or time.h... time.h
checking size of int... 4
checking size of short... 2
checking size of long... 8
checking for stdlib.h... (cached) yes
checking for GNU libc compatible malloc... yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible realloc... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for CHECK... yes
checking for CPPUNIT... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating libsubunit.pc
config.status: creating libcppunit_subunit.pc
config.status: creating Makefile
config.status: creating perl/Makefile.PL
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands

----Running------
> make -j8
-----------------
make  all-am
make[1]: Entering directory '/home/maarten/.conan/data/subunit/1.4.0/_/_/build/bff1fe8d4fd123419ee54b7d4fcc1b4dea9ea1db'
mkdir -p perl
depbase=`echo c++/SubunitTestProgressListener.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ./libtool  --tag=CXX   --mode=compile /usr/bin/g++ -DHAVE_CONFIG_H -I. -Isource_subfolder   -I/home/maarten/.conan/data/cppunit/1.15.1/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/include -DNDEBUG -D_GLIBCXX_USE_CXX11_ABI=1 -Wall -Werror -Wextra -Wwrite-strings -Wno-variadic-macros -Isource_subfolder/c/include -m64 -O3 -s -fPIC -MT c++/SubunitTestProgressListener.lo -MD -MP -MF $depbase.Tpo -c -o c++/SubunitTestProgressListener.lo source_subfolder/c++/SubunitTestProgressListener.cpp &&\
mv -f $depbase.Tpo $depbase.Plo
cd perl && perl Makefile.PL INSTALLDIRS=site
depbase=`echo c/lib/child.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ./libtool  --tag=CC   --mode=compile /usr/bin/gcc -DHAVE_CONFIG_H -I. -Isource_subfolder   -I/home/maarten/.conan/data/cppunit/1.15.1/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/include -DNDEBUG -D_GLIBCXX_USE_CXX11_ABI=1 -Wall -Werror -Wextra -Wstrict-prototypes  -Wmissing-prototypes -Wwrite-strings  -Wno-variadic-macros  -Isource_subfolder/c/include -m64 -O3 -s -fPIC -MT c/lib/child.lo -MD -MP -MF $depbase.Tpo -c -o c/lib/child.lo source_subfolder/c/lib/child.c &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  /usr/bin/gcc -DHAVE_CONFIG_H -I. -Isource_subfolder -I/home/maarten/.conan/data/cppunit/1.15.1/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/include -DNDEBUG -D_GLIBCXX_USE_CXX11_ABI=1 -Wall -Werror -Wextra -Wstrict-prototypes -Wmissing-prototypes -Wwrite-strings -Wno-variadic-macros -Isource_subfolder/c/include -m64 -O3 -s -fPIC -MT c/lib/child.lo -MD -MP -MF c/lib/.deps/child.Tpo -c source_subfolder/c/lib/child.c -o c/lib/child.o
libtool: compile:  /usr/bin/g++ -DHAVE_CONFIG_H -I. -Isource_subfolder -I/home/maarten/.conan/data/cppunit/1.15.1/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/include -DNDEBUG -D_GLIBCXX_USE_CXX11_ABI=1 -Wall -Werror -Wextra -Wwrite-strings -Wno-variadic-macros -Isource_subfolder/c/include -m64 -O3 -s -fPIC -MT c++/SubunitTestProgressListener.lo -MD -MP -MF c++/.deps/SubunitTestProgressListener.Tpo -c source_subfolder/c++/SubunitTestProgressListener.cpp -o c++/SubunitTestProgressListener.o
/bin/sh ./libtool  --tag=CC   --mode=link /usr/bin/gcc -Wall -Werror -Wextra -Wstrict-prototypes  -Wmissing-prototypes -Wwrite-strings  -Wno-variadic-macros  -Isource_subfolder/c/include -m64 -O3 -s -fPIC  -m64 -L/home/maarten/.conan/data/cppunit/1.15.1/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/lib -o libsubunit.la -rpath /home/maarten/.conan/data/subunit/1.4.0/_/_/package/bff1fe8d4fd123419ee54b7d4fcc1b4dea9ea1db/lib c/lib/child.lo  
Only one of PREFIX or INSTALL_BASE can be given.  Not both.
make[1]: *** [Makefile:1626: perl/Makefile] Error 2
make[1]: *** Waiting for unfinished jobs....
libtool: link: ar cru .libs/libsubunit.a  c/lib/child.o
libtool: link: ranlib .libs/libsubunit.a
libtool: link: ( cd ".libs" && rm -f "libsubunit.la" && ln -s "../libsubunit.la" "libsubunit.la" )
make[1]: Leaving directory '/home/maarten/.conan/data/subunit/1.4.0/_/_/build/bff1fe8d4fd123419ee54b7d4fcc1b4dea9ea1db'
make: *** [Makefile:696: all] Error 2

```
</details>

Reply:
I dropped the patch.
This will make the package not build on my machine, but it succeeds on a docker.
So let's see what c3i says.

Reply:
Great!

Only a few macOS builds fails with a strange perl error.
I checked my Xcode 12 installation and there is a file '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE/config.h' the build complaints of. Just checked dists of Xcode 9.4.1 and CLI tools - there is no such directory either. Maybe it is created during actual installation as someone suggested on StackOverflow.

Reply:
Wondering how many macOS builds are affected by this change - seems like only xcode 12. We'll see after merging (it's hard to see in C3 logs)